engine/src/main/java/org/terasology/config/flexible/FlexibleConfigImpl.java

```diff
@@ -52,7 +55,14 @@ public boolean add(Setting setting) {
             return false;
         }
 
+        if (unusedSettings != null) {
+            if (unusedSettings.containsKey(id)) {
+                setting.setValueFromString(unusedSettings.remove(id));
+            }
+        }
```

Comment:
This could be an unexpected sideeffect because a caller of `fc.add(setting)` could expect that an add call keeps the value of the setting the same.

Reply:
I'm not sure how else we can handle the saving/loading part.

We can make it clear in the javadocs that adding a settings will loads the non-default value that was saved earlier, and that if you want the default value, you must call `.resetValue()` right after?

Reply:
I guess you can, but I don't think we should encourage anybody to override a user's choice without his or her consent.

Reply:
My previous comment wasn't quite concrete enough. I'd say: yes document the behaviour in the javadoc, no do not suggest to call resetValue() if necessary - it shouldn't be necessary unless the user wants it.